src/Compiler/Driver/GraphChecking/GraphProcessing.fs

```diff
@@ -231,7 +231,11 @@ let processGraphAsync<'Item, 'Result when 'Item: equality and 'Item: comparison>
         let processedCount = IncrementableInt(0)
 
         let raiseExn (item, ex: exn) =
-            localCts.Cancel()
+            try
```

Comment:
This only handles the one edge case where it gets called simultaneously from multiple threads and first one causes the upper scope to finish and dispose. (At least that's what I believe caused this to crash the tests.)

I don't see how this could be handled from a level below. At least without making it more complicated.

Reply:
Well this is not a call to action, rather just a note that this function now does more than just "raising exceptions", hence an assumption that this is not the right place. But we can live with that

Reply:
Well it did that before already :) Here we just add a guard.

Also if it just raised exception, then you wouldn't really need it. This is more "things that we should do when an exception is raised". Maybe it should be renamed to `handleException`?

Reply:
Yeah, good idea.

Reply:
There's a lot of similar looking code in FSAC handing this exact stupid case 😂